src/test/java/io/logz/jmx2graphite/GraphiteITest.java

```diff
@@ -0,0 +1,139 @@
+package io.logz.jmx2graphite;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static io.restassured.RestAssured.given;
+import static java.time.temporal.ChronoUnit.MINUTES;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.io.File;
+import java.net.ConnectException;
+import java.time.Duration;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.BindMode;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.Lists;
+
+@SuppressWarnings({ "SpellCheckingInspection", "rawtypes" })
+public class GraphiteITest {
+
+    private final static Logger logger = LoggerFactory.getLogger(GraphiteITest.class);
+
+    private static GenericContainer graphiteContainer;
+    private static Stopwatch beforeClassStopwatch;
+    public static final String TEST_SERVICE_HOST = "testhost";
+    public static final String TEST_SERVICE_NAME = "testservice";
+    private static File tempDir;
+
+    @BeforeClass
+    public static void beforeClass() {
+        beforeClassStopwatch = Stopwatch.createStarted();
+        tempDir = createTempDir();
+
+        graphiteContainer = new GenericContainer("sitespeedio/graphite:0.9.14").withExposedPorts(2003, 80)
+                .withFileSystemBind(tempDir.getAbsolutePath(), "/var/log/carbon", BindMode.READ_WRITE)
+                .withClasspathResourceMapping("carbon.conf", "/opt/graphite/conf/carbon.conf", BindMode.READ_ONLY)
+                .withClasspathResourceMapping("blacklist.conf", "/opt/graphite/conf/blacklist.conf", BindMode.READ_ONLY)
+                .withClasspathResourceMapping("storage-schemas.conf", "/opt/graphite/conf/storage-schemas.conf",
+                        BindMode.READ_ONLY)
+                .waitingFor(new HostPortWaitStrategy().withStartupTimeout(Duration.ofSeconds(10)))
+                .withLogConsumer(new Slf4jLogConsumer(logger).withPrefix("graphite"));
+
+        graphiteContainer.setPortBindings(Lists.newArrayList("2003:2003", "1180:80"));
+        graphiteContainer.start();
+        beforeClassStopwatch.stop();
+    }
+
+    private static File createTempDir() {
+        File tempDir = new File("/tmp/mytest" + System.currentTimeMillis());
```

Comment:
Maybe it's better to have it more generic so to speak:
```java
        final File tempFile = File.createTempFile("mytest", "");
        if (!tempFile.delete()) throw new RuntimeException("Failed deleting temporary directory ("+tempFile.getAbsolutePath()+") while creating a temp dir");
        if (!tempFile.mkdirs()) throw new RuntimeException("Failed creating temporary directory: "+tempFile.getAbsolutePath());;
```

I'm using createTempFile for both: creating it in the right place, platform wise, and also name generation. then deleting created file and using that name to create a directory which in effect a temp directory

while we're at it, let's move this private function down below.

Reply:
Awesome, did not even know that method existed, thanks. I also moved the method to the bottom of the file.